include/dr/mhp/algorithms/copy.hpp

```diff
@@ -61,4 +62,26 @@ void copy(std::size_t root, rng::contiguous_range auto &&in,
   barrier();
 }
 
+// copy for dr::views::iota
```

Comment:
A copy with root for iota is not needed and this implementation would be very slow. It would be better to generalize the constraints on the first argument type here: https://github.com/oneapi-src/distributed-ranges/blob/fd4ca869190e32fa75ce9e73edd3a23ad8cc47b6/include/dr/mhp/algorithms/copy.hpp#L13